### ISSUE TITLE
Toolkit plugin crashing with "random" stack traces on both Mac and Linux

### ISSUE THREAD

**Issue by Marakai** [aws/aws-toolkit-jetbrains#1094]
**Describe the bug**

After a recent update to the latest  build of PyCharm and updating to AWS tookit 1.4 at the same time, AWS serverless apps have started to crash with the stack trace(!)below. They occur in both Linux and MacOS - sometimes one, some times the other.Apologies that I can be more specific - it's that random!

 I've tested it with something as simple as the Hello World app - where is will *sometimes* still work (crashes about half the time) - and my own, which deploys two Lambdas in its template.yaml.

**To reproduce**
0. Run repeated PIP to make sure all your packages, especially the AWS ones (awscli, aws-sam-cli, boto3, botocore and dependencies) are the latest
1. Start PyCharm
2. Load project
3. Set Run Configuration
4. Debug

**Your Environment**
 - OS: CentOS 7.5 and MacOS 10.14.5
 - JetBrains' Product: PyCharm
 - JetBrains' Product Version: 2019.1.3 191.7479.30
 - Toolkit Version: 1.4
 - SAM CLI Version: 0.17.0
 - JVM/Python Version: 1.8.0_201-b09

**Additional context**

template.yaml (for my project, though the default for Hello World could be shown here)

```
AWSTemplateFormatVersion: '2010-09-09'
Transform: AWS::Serverless-2016-10-31
Description: Symantec MSS Configuration

Globals:
  Function:
    Timeout: 3

Resources:
  GuardDutyFunction:
    Type: AWS::Serverless::Function
    Properties:
      CodeUri: guardduty/
      Handler: app.lambda_handler
      Runtime: python3.7

  SymantecS3Function:
    Type: AWS::Serverless::Function
    Properties:
      CodeUri: s3/
      Handler: app.lambda_handler
      Runtime: python3.7

  EventRuleGuardDuty:
    Type: AWS::Events::Rule
    Properties:
      Description: "EventRule"
      EventPattern:
        source:
          - "aws.guardduty"
        detail-type:
          - "GuardDuty Finding"
      State: "ENABLED"
      Targets:
        -
          Arn:
            Fn::GetAtt:
              - "GuardDutyFunction"
              - "Arn"
          Id: "GuardDutyTargetFunctionV1"

  PermissionForEventsToInvokeLambdaGuardDuty:
    Type: AWS::Lambda::Permission
    Properties:
      FunctionName:
        Ref: "GuardDutyFunction"
      Action: "lambda:InvokeFunction"
      Principal: "events.amazonaws.com"
      SourceArn:
        Fn::GetAtt:
          - "EventRuleGuardDuty"
          - "Arn"

  EventRuleS3:
    Type: AWS::Events::Rule
    Properties:
      Description: "EventRule"
      EventPattern:
        source:
          - "aws.s3"
        detail-type:
          - "AWS API call via CloudTrail"
        detail:
          eventSource: s3.awsamazon.com
          eventName:
            - CreateBucket
            - PutBucketAcl
      State: "ENABLED"
      Targets:
        - Arn:
            Fn::GetAtt:
              - "SymantecS3Function"
              - "Arn"
          Id: "TargetFunctionV1"

  PermissionForEventsToInvokeLambdaS3:
    Type: AWS::Lambda::Permission
    Properties:
      FunctionName:
        Ref: "SymantecS3Function"
      Action: "lambda:InvokeFunction"
      Principal: "events.amazonaws.com"
      SourceArn:
        Fn::GetAtt:
          - "EventRuleS3"
          - "Arn"

Outputs:
  GuardDutyFunction:
    Description: "Symantec Guardduty Lambda Function ARN"
    Value: !GetAtt GuardDutyFunction.Arn
  GuardDutyFunctionIamRole:
    Description: "Implicit IAM Role created for Symantec Guardduty function"
    Value: !GetAtt GuardDutyFunctionRole.Arn
  SymantecS3Function:
    Description: "Symantec S3 Function ARN"
    Value: !GetAtt SymantecS3Function.Arn
  SymantecS3FunctionIamRole:
    Description: "Implicit IAM Role created for Symantec S3 function"
    Value: !GetAtt SymantecS3FunctionRole.Arn


```
Stack traces:

Some times I get this one

```

java.lang.Throwable: Assertion failed: Network calls can't be made inside read/write action
	at software.aws.toolkits.core.utils.LogUtilsKt.assertTrue(LogUtils.kt:91)
	at software.aws.toolkits.jetbrains.core.AwsSdkClient$ValidateCorrectThreadClient.prepareRequest(AwsSdkClient.kt:46)
	at software.amazon.awssdk.core.client.builder.SdkDefaultClientBuilder$NonManagedSdkHttpClient.prepareRequest(SdkDefaultClientBuilder.java:386)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.MakeHttpRequestStage.executeHttpRequest(MakeHttpRequestStage.java:59)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.MakeHttpRequestStage.execute(MakeHttpRequestStage.java:51)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.MakeHttpRequestStage.execute(MakeHttpRequestStage.java:35)
	at software.amazon.awssdk.core.internal.http.pipeline.RequestPipelineBuilder$ComposingRequestPipelineStage.execute(RequestPipelineBuilder.java:206)
	at software.amazon.awssdk.core.internal.http.pipeline.RequestPipelineBuilder$ComposingRequestPipelineStage.execute(RequestPipelineBuilder.java:206)
	at software.amazon.awssdk.core.internal.http.pipeline.RequestPipelineBuilder$ComposingRequestPipelineStage.execute(RequestPipelineBuilder.java:206)
	at software.amazon.awssdk.core.internal.http.pipeline.RequestPipelineBuilder$ComposingRequestPipelineStage.execute(RequestPipelineBuilder.java:206)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.ApiCallAttemptTimeoutTrackingStage.execute(ApiCallAttemptTimeoutTrackingStage.java:64)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.ApiCallAttemptTimeoutTrackingStage.execute(ApiCallAttemptTimeoutTrackingStage.java:36)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.TimeoutExceptionHandlingStage.execute(TimeoutExceptionHandlingStage.java:77)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.TimeoutExceptionHandlingStage.execute(TimeoutExceptionHandlingStage.java:39)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.RetryableStage$RetryExecutor.doExecute(RetryableStage.java:113)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.RetryableStage$RetryExecutor.execute(RetryableStage.java:86)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.RetryableStage.execute(RetryableStage.java:62)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.RetryableStage.execute(RetryableStage.java:42)
	at software.amazon.awssdk.core.internal.http.pipeline.RequestPipelineBuilder$ComposingRequestPipelineStage.execute(RequestPipelineBuilder.java:206)
	at software.amazon.awssdk.core.internal.http.StreamManagingStage.execute(StreamManagingStage.java:57)
	at software.amazon.awssdk.core.internal.http.StreamManagingStage.execute(StreamManagingStage.java:37)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.ApiCallTimeoutTrackingStage.executeWithTimer(ApiCallTimeoutTrackingStage.java:80)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.ApiCallTimeoutTrackingStage.execute(ApiCallTimeoutTrackingStage.java:60)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.ApiCallTimeoutTrackingStage.execute(ApiCallTimeoutTrackingStage.java:42)
	at software.amazon.awssdk.core.internal.http.pipeline.RequestPipelineBuilder$ComposingRequestPipelineStage.execute(RequestPipelineBuilder.java:206)
	at software.amazon.awssdk.core.internal.http.pipeline.RequestPipelineBuilder$ComposingRequestPipelineStage.execute(RequestPipelineBuilder.java:206)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.ExecutionFailureExceptionReportingStage.execute(ExecutionFailureExceptionReportingStage.java:37)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.ExecutionFailureExceptionReportingStage.execute(ExecutionFailureExceptionReportingStage.java:26)
	at software.amazon.awssdk.core.internal.http.AmazonSyncHttpClient$RequestExecutionBuilderImpl.execute(AmazonSyncHttpClient.java:240)
	at software.amazon.awssdk.core.client.handler.BaseSyncClientHandler.invoke(BaseSyncClientHandler.java:96)
	at software.amazon.awssdk.core.client.handler.BaseSyncClientHandler.execute(BaseSyncClientHandler.java:120)
	at software.amazon.awssdk.core.client.handler.BaseSyncClientHandler.execute(BaseSyncClientHandler.java:73)
	at software.amazon.awssdk.core.client.handler.SdkSyncClientHandler.execute(SdkSyncClientHandler.java:44)
	at software.amazon.awssdk.awscore.client.handler.AwsSyncClientHandler.execute(AwsSyncClientHandler.java:55)
	at software.amazon.awssdk.services.sts.DefaultStsClient.assumeRole(DefaultStsClient.java:220)
	at software.amazon.awssdk.services.sts.auth.StsAssumeRoleCredentialsProvider.getUpdatedCredentials(StsAssumeRoleCredentialsProvider.java:68)
	at software.amazon.awssdk.services.sts.auth.StsCredentialsProvider.updateSessionCredentials(StsCredentialsProvider.java:68)
	at software.amazon.awssdk.utils.cache.CachedSupplier.refreshCache(CachedSupplier.java:132)
	at software.amazon.awssdk.utils.cache.CachedSupplier.get(CachedSupplier.java:89)
	at software.amazon.awssdk.services.sts.auth.StsCredentialsProvider.resolveCredentials(StsCredentialsProvider.java:78)
	at software.amazon.awssdk.services.sts.auth.StsAssumeRoleCredentialsProvider.resolveCredentials(StsAssumeRoleCredentialsProvider.java:42)
	at software.aws.toolkits.jetbrains.core.credentials.profiles.ProfileToolkitCredentialsProvider.resolveCredentials(ProfileToolkitCredentialsProvider.kt:53)
	at software.aws.toolkits.jetbrains.services.lambda.execution.local.SamRunningState.startProcess(SamRunningState.kt:30)
	at com.intellij.execution.configurations.CommandLineState.execute(CommandLineState.java:61)
	at software.aws.toolkits.jetbrains.services.lambda.python.PythonSamDebugSupport$createDebugProcess$1.start(PythonSamDebugSupport.kt:54)
	at com.intellij.xdebugger.impl.XDebuggerManagerImpl.startSession(XDebuggerManagerImpl.java:234)
	at com.intellij.xdebugger.impl.XDebuggerManagerImpl.startSession(XDebuggerManagerImpl.java:195)
	at software.aws.toolkits.jetbrains.services.lambda.execution.local.SamDebugger.run(SamDebugger.kt:36)
	at software.aws.toolkits.jetbrains.services.lambda.execution.local.SamInvokeRunner$execute$1$$special$$inlined$runInEdt$1.run(actions.kt:75)
	at com.intellij.openapi.application.TransactionGuardImpl$2.run(TransactionGuardImpl.java:315)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.doRun(LaterInvocator.java:435)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.runNextEvent(LaterInvocator.java:419)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.run(LaterInvocator.java:403)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(Unknown Source)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(Unknown Source)
	at java.desktop/java.awt.EventQueue$4.run(Unknown Source)
	at java.desktop/java.awt.EventQueue$4.run(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
	at java.desktop/java.awt.EventQueue.dispatchEvent(Unknown Source)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:729)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:678)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:373)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.run(Unknown Source)

```

and also some times (such as right now):

```
java.lang.AssertionError: Found 0 YAML files but only expected 1
	at software.aws.toolkits.jetbrains.services.lambda.sam.SamCommon$Companion.getTemplateFromDirectory(SamCommon.kt:148)
	at software.aws.toolkits.jetbrains.ui.wizard.SamProjectTemplate.createRunConfigurations(SamInitProjectBuilderCommon.kt:58)
	at software.aws.toolkits.jetbrains.ui.wizard.SamProjectTemplate.postCreationAction(SamInitProjectBuilderCommon.kt:47)
	at software.aws.toolkits.jetbrains.ui.wizard.SamPythonProjectTemplate.postCreationAction(SamInitProjects.kt:73)
	at software.aws.toolkits.jetbrains.ui.wizard.SamProjectBuilder.runPostModuleCreationStep(SamProjectGeneratorIntelliJShims.kt:97)
	at software.aws.toolkits.jetbrains.ui.wizard.SamProjectBuilder.setupRootModel(SamProjectGeneratorIntelliJShims.kt:65)
	at software.aws.toolkits.jetbrains.ui.wizard.SamProjectGenerator$generateProject$$inlined$runInEdt$lambda$1.compute(actions.kt:77)
	at com.intellij.openapi.application.impl.ApplicationImpl.lambda$runWriteAction$19(ApplicationImpl.java:1064)
	at com.intellij.openapi.application.impl.ApplicationImpl.runWriteActionWithClass(ApplicationImpl.java:1020)
	at com.intellij.openapi.application.impl.ApplicationImpl.runWriteAction(ApplicationImpl.java:1064)
	at software.aws.toolkits.jetbrains.ui.wizard.SamProjectGenerator.generateProject(SamProjectGenerator.kt:157)
	at software.aws.toolkits.jetbrains.ui.wizard.SamProjectGenerator.generateProject(SamProjectGenerator.kt:43)
	at com.intellij.ide.util.projectWizard.AbstractNewProjectStep.lambda$doGenerateProject$2(AbstractNewProjectStep.java:202)
	at com.intellij.platform.PlatformProjectOpenProcessor.doOpenProject(PlatformProjectOpenProcessor.java:266)
	at com.intellij.ide.util.projectWizard.AbstractNewProjectStep.doGenerateProject(AbstractNewProjectStep.java:207)
	at com.intellij.ide.util.projectWizard.AbstractNewProjectStep$AbstractCallback.consume(AbstractNewProjectStep.java:161)
	at com.intellij.ide.util.projectWizard.ProjectSettingsStepBase$1.actionPerformed(ProjectSettingsStepBase.java:135)
	at java.desktop/javax.swing.AbstractButton.fireActionPerformed(Unknown Source)
	at java.desktop/javax.swing.AbstractButton$Handler.actionPerformed(Unknown Source)
	at java.desktop/javax.swing.DefaultButtonModel.fireActionPerformed(Unknown Source)
	at java.desktop/javax.swing.DefaultButtonModel.setPressed(Unknown Source)
	at java.desktop/javax.swing.plaf.basic.BasicButtonListener.mouseReleased(Unknown Source)
	at java.desktop/java.awt.Component.processMouseEvent(Unknown Source)
	at java.desktop/javax.swing.JComponent.processMouseEvent(Unknown Source)
	at java.desktop/java.awt.Component.processEvent(Unknown Source)
	at java.desktop/java.awt.Container.processEvent(Unknown Source)
	at java.desktop/java.awt.Component.dispatchEventImpl(Unknown Source)
	at java.desktop/java.awt.Container.dispatchEventImpl(Unknown Source)
	at java.desktop/java.awt.Component.dispatchEvent(Unknown Source)
	at java.desktop/java.awt.LightweightDispatcher.retargetMouseEvent(Unknown Source)
	at java.desktop/java.awt.LightweightDispatcher.processMouseEvent(Unknown Source)
	at java.desktop/java.awt.LightweightDispatcher.dispatchEvent(Unknown Source)
	at java.desktop/java.awt.Container.dispatchEventImpl(Unknown Source)
	at java.desktop/java.awt.Window.dispatchEventImpl(Unknown Source)
	at java.desktop/java.awt.Component.dispatchEvent(Unknown Source)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(Unknown Source)
	at java.desktop/java.awt.EventQueue$4.run(Unknown Source)
	at java.desktop/java.awt.EventQueue$4.run(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
	at java.desktop/java.awt.EventQueue$5.run(Unknown Source)
	at java.desktop/java.awt.EventQueue$5.run(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
	at java.desktop/java.awt.EventQueue.dispatchEvent(Unknown Source)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:729)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:674)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:373)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.run(Unknown Source)
```

**Comment by Marakai** [comment]
Brief addendum:

While updating the above I had also run a full scale PIP on all packages - so it's somewhat hard to track down if any specific one could be a culprit, provided it IS a Python dependency that's causing this.

**Comment by abrooksv** [comment]
The first call is just a warning, the call does go through but is a warning to us to fix it so we do not degrade the IDE experience

The second one is hard to tell given I am not sure what template you ran from the project wizard, but we did fix some threading issues and an incompatibility with the dynamoDB template.

Both of these issues have been fixed in our 1.5 EAP builds.

**Comment by Marakai** [comment]
I tried both the Hello World and the DynamoDB base templates, just to cover all bases.

**Comment by abrooksv** [comment]
Ya, the DynamoDB one is guaranteed to fail in 1.4. We have fixed this in master and is pending the 1.5 release. You can opt-in to EAP builds if you wish. (See the README)

The Hello World is random due to the threading issues. It only prevents marking the hello_world folder as a source root. If you do this manually there is no difference. This is also fixed in 1.5 already.

The call for the network thread ("Assertion failed: Network calls..") can be ignored, it doesn't harm you, it just helps us by having people report the issue so we can make sure the call is fixed to not be on the UI thread and block your IDE. We have not fixed this instance yet of the issue. I will try to squeeze the fix into the 1.5 release, but no guarantee.

Thank you for the detailed bug reports and triaging.

**Comment by abrooksv** [comment]
Fixed in 1.5, it is pending JetBrains' approval on marketplace
